gateway-server/src/main/java/org/apache/knox/gateway/services/factory/ConcurrentSessionVerifierFactory.java

```diff
@@ -17,21 +17,37 @@
  */
 package org.apache.knox.gateway.services.factory;
 
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+
+import java.util.Collection;
+import java.util.Map;
+
 import org.apache.knox.gateway.config.GatewayConfig;
 import org.apache.knox.gateway.services.GatewayServices;
 import org.apache.knox.gateway.services.Service;
 import org.apache.knox.gateway.services.ServiceLifecycleException;
 import org.apache.knox.gateway.services.ServiceType;
+import org.apache.knox.gateway.session.control.EmptyConcurrentSessionVerifier;
 import org.apache.knox.gateway.session.control.InMemoryConcurrentSessionVerifier;
 
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Map;
-
 public class ConcurrentSessionVerifierFactory extends AbstractServiceFactory {
   @Override
   protected Service createService(GatewayServices gatewayServices, ServiceType serviceType, GatewayConfig gatewayConfig, Map<String, String> options, String implementation) throws ServiceLifecycleException {
-    return shouldCreateService(implementation) ? new InMemoryConcurrentSessionVerifier() : null;
+    Service service = null;
+    if (shouldCreateService(implementation)) {
```

Comment:
If `shouldCreateService` returns false we don't create any implementation and we return null? Later at other place we have assumption that there is always an implementation available (either the null object or the real one).

Reply:
As far as i can see, it is impossible to add null as a service. When we add the service with the `addService` method of `AbstractGatewayServices` class a `NullPointerException` is thrown by these lines:
```
// Ensure the service is not null
if (service == null) {
   throw new NullPointerException("A null service may not be added.");
}
```

Reply:
I agree with @MrtnBalazs on this. It's perfectly ok to return a `null` service from any of the service factories. That situation is handled by the caller.